bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Modal.java

```diff
@@ -130,8 +135,24 @@ protected void populateItem(ListItem<Component> item) {
 
         BootstrapResourcesBehavior.addTo(this);
     }
-
+    
     /**
+     * Factory method for the Modal's header 'closebutton'
+     * By default it generates the markup for original Bootstrap Modal closebutton
+     * 
+     * @param id The component id
+     * @return the close button
+     */
+    
+    protected Component createHeaderCloseButton(String id) {
+
+    	BootstrapButton btn = new BootstrapButton(id, Buttons.Type.Link);
+    	btn.setLabel(new  Model<String>("\u00D7"));
+    	btn.add(new AttributeModifier("class","close"));
```

Comment:
I wanted to retain the original markup. The button alone will change the markup. There's also css at play. If you don't override the button the modal will have no correct markup. I 'll try to post a quickstart when i come home.
cheers